featuretools/entityset/entityset.py

```diff
@@ -280,231 +294,240 @@ def add_relationship(self,
-            parent_e.convert_variable_type(variable_id=parent_v,
-                                           new_type=vtypes.Index,
-                                           convert_data=False)
+            raise ValueError(msg.format(child_column, child_df.ww.name))
+        parent_df = relationship.parent_dataframe
+        parent_column = relationship._parent_column_id
+        if 'foreign_key' not in child_df.ww.semantic_tags[child_column]:
+            child_df.ww.add_semantic_tags({child_column: 'foreign_key'})
+
+        if parent_df.ww.index != parent_column:
+            parent_df.ww.set_index(parent_column)
         # Empty dataframes (as a result of accessing Entity.metadata)
         # default to object dtypes for discrete variables, but
         # indexes/ids default to ints. In this case, we convert
         # the empty column's type to int
-        if isinstance(child_e.df, pd.DataFrame) and \
-                (child_e.df.empty and child_e.df[child_v].dtype == object and
-                 is_numeric_dtype(parent_e.df[parent_v])):
-            child_e.df[child_v] = pd.Series(name=child_v, dtype=np.int64)
-
-        parent_dtype = parent_e.df[parent_v].dtype
-        child_dtype = child_e.df[child_v].dtype
-        msg = u"Unable to add relationship because {} in {} is Pandas dtype {}"\
-            u" and {} in {} is Pandas dtype {}."
-        if not is_dtype_equal(parent_dtype, child_dtype):
-            raise ValueError(msg.format(parent_v, parent_e.id, parent_dtype,
-                                        child_v, child_e.id, child_dtype))
+        # --> Implementation: is this still relevant?
+        if isinstance(child_df, pd.DataFrame) and \
+                (child_df.empty and child_df[child_column].dtype == object and
+                 parent_df.ww.columns[parent_column].is_numeric):
+            child_df.ww[child_column] = pd.Series(name=child_column, dtype=np.int64)
+
+        parent_ltype = parent_df.ww.logical_types[parent_column]
+        child_ltype = child_df.ww.logical_types[child_column]
+        if parent_ltype != child_ltype:
+            warnings.warn(f'Logical type {child_ltype} for child column {child_column} does not match '
+                          f'parent column {parent_column} logical type {parent_ltype}. '
+                          'Changing child logical type to match parent.')
+            child_df.ww.set_types(logical_types={child_column: parent_ltype})
 
         self.relationships.append(relationship)
         self.reset_data_description()
         return self
 
-    def set_secondary_time_index(self, entity, secondary_time_index):
-        self._check_secondary_time_index(entity, secondary_time_index)
-        entity.secondary_time_index = secondary_time_index
+    def set_secondary_time_index(self, dataframe_id, secondary_time_index):
+        '''Sets the secondary time index for a dataframe in the EntitySet using its dataframe id'''
```

Comment:
Can we do a longer docstring that includes the Args format

Reply:
updated!